sample-code/src/orchestration.ts

```diff
@@ -633,3 +632,38 @@ export async function orchestrationMessageHistoryWithToolCalling(): Promise<Orch
     messagesHistory: allMessages
   });
 }
+
+/**
+ * Ask ChatGPT to do add two numbers with tools.
+ * @param controller - The abort controller.
+ * @param streamOptions - The stream options.
+ * @returns The response from the orchestration service containing the response content.
+ */
+export async function chatCompletionStreamWithTools(
+  controller: AbortController,
+  streamOptions?: StreamOptions
+): Promise<OrchestrationStreamResponse<OrchestrationStreamChunkResponse>> {
+  const orchestrationClient = new OrchestrationClient({
+    // define the language model to be used
+    llm: {
+      model_name: 'gpt-4o'
+    },
+    // define the prompt
+    templating: {
+      template: [
```

Comment:
[req] With the `template` property change, maybe we can now use messages. This use case doesnt seem like it needs a template